core/shared/src/main/scala/zio/internal/Platform.scala

```diff
@@ -77,17 +78,47 @@ trait Platform { self =>
    */
   def reportFailure(cause: Cause[Any]): Unit
 
+  /**
+   * Print Fiber Dump of all fibers when interrupt signal received
+   */
+  def onInterruptSignal: Platform =
```

Comment:
We can't expose this here because we need a `Runtime` that can unsafely run the effect. Instead we will implement this on `Runtime` and call `onInterruptSignal`, using the runtime to unsafely run the effect in the handler.